### ISSUE TITLE
2.2.17 breaks building with docker provider

### ISSUE THREAD

**Issue by vafschoen** [hashicorp/vagrant#12447]
### Vagrant version
Vagrant 2.2.17

### Host operating system
Ubuntu 21.10, 21.04, 20.04

### Guest operating system
n.a.

### Vagrantfile

```ruby
Vagrant.configure("2") do |config|
  config.vm.provider "docker" do |d|
    d.build_dir = "."
  end
end
```

### Dockerfile
```
FROM nginx
```

### Debug output
https://gist.github.com/vafschoen/a99aa80930f5826bbaff5d68510f75c1

### Expected behavior

Start building a container

### Actual behavior

Vagrant exits with
```/opt/vagrant/embedded/gems/2.2.17/gems/childprocess-4.1.0/lib/childprocess/abstract_process.rb:44:in `initialize': all arguments must be String: ["/usr/bin/docker", "build", "/home/%USER%/tmp/vagrant-2.2.17", {:with_stderr=>true}] (ArgumentError)```

### Steps to reproduce

1. vagrant up

**Comment by gabriel-amaral** [comment]
Same problem here. Host Mac

**Comment by denisacostaq** [comment]
same problem on my side
Vagrant 2.2.17
Linux debian 4.19.0-17-amd64 #1 SMP Debian 4.19.194-1 (2021-06-10) x86_64 GNU/Linux

**Comment by vishesh-deepsource** [comment]
Facing the same problem on a an M1 Apple device. Do you have an estimate on when this fix will be released?

**Comment by deatinor** [comment]
Same here on Mac M1

**Comment by ashvarma1982** [comment]
is there a fix for this bug? I am also facing the same
```
vagrant up --provider=docker
Bringing machine 'default' up with 'docker' provider...
==> default: Creating and configuring docker networks...
==> default: Building the container from a Dockerfile...
/opt/vagrant/embedded/gems/2.2.17/gems/childprocess-4.1.0/lib/childprocess/abstract_process.rb:44:in `initialize': all arguments must be String: ["/usr/local/bin/docker", "build", "/Users/ashish.varma/workspace/VMs/docker_vagrant_ubuntu", {:with_stderr=>true}] (ArgumentError)
	from /opt/vagrant/embedded/gems/2.2.17/gems/childprocess-4.1.0/lib/childprocess.rb:23:in `new'
	from /opt/vagrant/embedded/gems/2.2.17/gems/childprocess-4.1.0/lib/childprocess.rb:23:in `new'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/subprocess.rb:84:in `execute'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/subprocess.rb:22:in `execute'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/executor/local.rb:17:in `block in execute'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/executor/local.rb:16:in `execute'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/driver.rb:315:in `execute'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/driver.rb:26:in `build'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/action/build.rb:69:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/action/login.rb:32:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/action/prepare_networks.rb:364:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/action/prepare_nfs_settings.rb:15:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/synced_folder_cleanup.rb:28:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/synced_folders/nfs/action_cleanup.rb:25:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/action/prepare_nfs_valid_ids.rb:14:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/action/host_machine_sync_folders.rb:23:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/action/host_machine_build_dir.rb:26:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/env_set.rb:19:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builder.rb:149:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `block in run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/call.rb:53:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/provision.rb:80:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builder.rb:149:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `block in run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/call.rb:53:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builder.rb:149:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `block in run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/call.rb:53:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builder.rb:149:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `block in run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/call.rb:53:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/plugins/providers/docker/action/host_machine.rb:18:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:127:in `block in finalize_action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builder.rb:149:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `block in run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builtin/call.rb:53:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/warden.rb:48:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/builder.rb:149:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `block in run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/action/runner.rb:89:in `run'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/machine.rb:246:in `action_raw'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/machine.rb:215:in `block in action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/environment.rb:614:in `lock'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/machine.rb:201:in `call'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/machine.rb:201:in `action'
	from /opt/vagrant/embedded/gems/2.2.17/gems/vagrant-2.2.17/lib/vagrant/batch_action.rb:86:in `block (2 levels) in run'
```

**Comment by skoef** [comment]
The mentioned PR #12449 fixes the issue and is released in 2.2.18
